lib/graphql/lang/ast/type_info_visitor.ex

```diff
@@ -127,7 +127,7 @@ defmodule GraphQL.Lang.AST.TypeInfoVisitor do
         :ObjectField ->
           input_type = TypeInfo.input_type(accumulator[:type_info])
           object_type = TypeInfo.named_type(accumulator[:type_info], input_type)
-          if object_type === %GraphQL.Type.ObjectType{} do
+          if %GraphQL.Type.ObjectType{} = object_type do
```

Comment:
Yeah this code is quite dense. It's literally directly translated from the `graphql-js` implementation, which has no unit tests (it's tested indirectly via all the validation rule tests) which makes it hard to understand.